OrbitLinuxTracing/UprobesUnwindingVisitor.h

```diff
@@ -23,6 +26,34 @@ namespace LinuxTracing {
 // of instrumented functions. When we have a callstack broken because of
 // uretprobes we can then rebuild the missing part by joining together the parts
 // on the stack of callstacks associated with that thread.
+// TODO: Make this more robust to losing uprobes or uretprobes events (loss of
+//  uretprobes events should be rare if they don't come with a stack sample).
+//  Start by passing the function_address to ProcessUretprobes as well and for a
+//  comparison against the address of the uprobe on the stack.
+
+class UprobesFunctionCallManager {
+ public:
+  UprobesFunctionCallManager() = default;
+
+  UprobesFunctionCallManager(const UprobesFunctionCallManager&) = delete;
+  UprobesFunctionCallManager& operator=(const UprobesFunctionCallManager&) =
+      delete;
+
+  UprobesFunctionCallManager(UprobesFunctionCallManager&&) = default;
+  UprobesFunctionCallManager& operator=(UprobesFunctionCallManager&&) = default;
+
+  void ProcessUprobes(pid_t tid, uint64_t function_address,
+                      uint64_t begin_timestamp);
+  std::optional<FunctionCall> ProcessUretprobes(pid_t tid,
+                                                uint64_t end_timestamp);
+
+ private:
+  // This map keeps the stack of the dynamically-instrumented functions entered.
```

Comment:
This type is pretty unreadable. What do the uint64_t types represent? What to the pairs represent? These should be properly named types (structs).